File: api/ViaQuestInc.StepOne.Core/Candidates/Services/CandidateSearchRequest.cs
using ViaQuestInc.StepOne.Core.Candidates.Workflows;
using ViaQuestInc.StepOne.Core.Data;

namespace ViaQuestInc.StepOne.Core.Candidates.Services;

public class CandidateSearchRequest : SearchRequestBase
{
    public string? Name { get; set; }
    
    public int? CompanyId { get; set; }
    
    public int? JobTitleId { get; set; }

    public string? AtsJobTitle { get; set; }
    
    public CandidateWorkflowStatus? CandidateWorkflowStatus { get; set; }
}

File: ui/src/app/models/candidates/candidate-search.request.ts
import {SearchRequest} from "../search/search.request";
import {CandidateStatus} from "./candidate.status";
import {CandidateWorkflowStatuses} from "./candidate-workflow.statuses";

export interface CandidateSearchRequest extends SearchRequest {
  name?: string | null;

  candidateWorkflowStatus?: CandidateWorkflowStatuses | null;

  companyId?: number | null;

  jobTitleId?: number | null;

  atsJobTitle?: string | null;
}
